### DISCUSSION TITLE
EMQX node lost information about routing table

### DISCUSSION THREAD

**Discussion by on-delete** [emqx/emqx#8159] [General]
We have an emqx 4.3.11 cluster setup in kubernetes, with 3 nodes. We suddenly had some dropped messages (counted by emqx_messages_dropped_no_subscribers), but only for one node. On the two other nodes, the messages designated to the same topic, were delivered as expected. After a restart of all nodes, the problem was fixed.

The metric is indicating, that one node had no information about the subscription of the topic, even though the other two nodes had this information as it seems. Thats why it dropped these messages with reason 'no subscriber'. 

Could it be, that the one node had lost the information in his routing table? Could it be, that it was not replicated from another node initially? How could we detect such out of sync problems? The metrics are indicating, that all nodes had recognized all other nodes within the cluster during that time.

I did not wanted to post it as an issue, as we do not have any emqx dump from this incident. Logs also not indicating any failure during that time, that could have let to this problem.

**Comment by zmstone** [comment]
Hi @on-delete 
Do you have any logs kept before or around when it started happening?
Also, do you know how many topics were not synced?

4.3.x has a bug in the underlying Erlang virtual machine which may cause intra-cluster rpc timeouts (and hang indefinitely when timeout is set to infinity).
Please consider upgrading to v4.4.4.

**Reply by on-delete** [reply]
Hey @zmstone, thanks for your reply.

i looked into the logs more closely, and indeed it looks like one node had restarted and had no connection to the other nodes. It happened on may 29th, but our incident was at may 31th, thats why we didn't checked the earlier logs. I have attached the logs in a seperate file.
[emqx.log](https://github.com/emqx/emqx/files/8867180/emqx.log)

I checked the metric `emqx_cluster_nodes_running` during that time, and all 3 nodes have reported, that there are 3 nodes within the cluster running. Should we not at least see with this metric, if one node is not able to connect to the other nodes?

But we will now consider to upgrade to the newest emqx version, if this was caused by a bug. Thanks for the hint!
